exporter/sumologicexporter/compress.go

```diff
@@ -48,8 +49,10 @@ func newCompressor(format CompressEncodingType) (compressor, error) {
 		if err != nil {
 			return compressor{}, err
 		}
-	default:
+	case NoCompression:
 		writer = nil
+	default:
+		return compressor{}, fmt.Errorf("invalid format: %s", format)
```

Comment:
Can we use `github.com/pkg/errors` instead of `fmt`'s `Errorf`? Errors created by that package include more info about where the errors is coming from, e.g. stack traces

Other receivers and exporters in this repo already use it